Add Linear Cell Complex octree generation functionality

ybellargui opened the pull request:
## Summary of Changes

This PR implements octree generation functionality for CGAL's Linear Cell Complex package. It adds the ability to generate hexahedral octrees from OFF mesh files using AABB tree-based intersection testing. The implementation provides a simple 4-line API for easy integration and supports both Combinatorial Map and Generalized Map backends.

## Release Management

* Affected package(s): Linear_cell_complex
* Feature/Small Feature (if any): Small Feature - Linear Cell Complex octree generation
* Link to compiled documentation (obligatory for small feature): [Documentation will be generated after merge](https://doc.cgal.org/latest/Manual/packages.html#PkgLinearCellComplexSummary)
* License and copyright ownership: GPL v3+ / LGPL v3+ (same as CGAL), Copyright (c) 2025 CNRS and LIRIS' Establishments (France)

## Technical Details

**New API Function:**
```cpp
template<typename LCC>
void CGAL::compute_octree(LCC& lcc,
                         const std::string& off_filename,
                         unsigned int max_level = 5,
                         unsigned int init = 3,
                         bool create_all_voxels = false,
                         bool display_statistics = false,
                         bool use_binary_OFF = false);
```

**Files Added:**
- Linear_cell_complex/include/CGAL/Linear_cell_complex_octree_generation.h
- Linear_cell_complex/examples/Linear_cell_complex/lcc_octree_generation.cpp
- Linear_cell_complex/test/Linear_cell_complex/test_octree_generation.cpp

**Testing Status:**
- 12 unit tests passing (6 for CMap + 6 for GMap)
- Validates hexahedral topology, parameter effects, and error handling

This implementation is based on the distributed-hexahedral-subdivision research project by Guillaume Damiand.

ybellargui committed:
Add Linear Cell Complex octree generation functionality

- Implement CGAL::compute_octree() for hexahedral octree generation from OFF files
- Add example lcc_octree_generation.cpp and test test_octree_generation.cpp
- Add comprehensive test suite with 12 tests (CMap + GMap)
- Support configurable subdivision levels and octree parameters
- Based on distributed-hexahedral-subdivision research project

ybellargui committed:
Fix REUSE compliance: add missing headers

gdamiand commented:
Instead of taking a filename as second parameter, it is probably better to take a surface mesh (any model of facegraph).

ybellargui commented:
I tried to address your comments:

- Added a main overload that takes a Surface_mesh (any FaceGraph) instead of relying on a filename.
- Kept the filename overloads only as convenience wrappers.
- Added the regularized boolean parameter. Right now it’s only a stub (no balancing yet); the hook is there to implement 1‑irregular balancing (difference of at most one level between adjacent hexes) later.
- Added a simple type constraint so string literals don’t get treated as a FaceGraph.
- Updated tests and an example to cover both FaceGraph and filename cases.

ybellargui committed:
LCC octree: add FaceGraph overload + regularized flag stub, keep file overloads, tests & example updated

afabri commented:
What is the relationship to the [Octree package](https://doc.cgal.org/latest/Orthtree/group__PkgOrthtreeRef.html) ?

afabri commented:
please remove trailing whitespace as reported in the CI.

soesau committed:
removing trailing whitespaces

soesau committed:
adding named parameters
reducing dependencies

soesau committed:
updated package dependencies

sloriot commented:
@gdamiand what is the status of this PR? Can it be tested?
